### ISSUE TITLE
`um` context history

### ISSUE THREAD

**Issue by wgordon17** [promptops/cli#7]
Just wondering if there are plans to have `um` store context about previous commands? 

Ahh, actually I see it in the help output (`--history-context`) and it defaults to `0`. So maybe the enhancement could be parse a _written_ command as opposed to a command line flag.

E.g., when a question starts with "in my last question", or "using the previous output", or something. Mostly just a nice to have, since it's perhaps a bit more intuitive vs a command line flag ¯\\\_(ツ)_/¯

**Comment by wgordon17** [comment]
Actually, it's not clear to me what `--history-context` is doing. Does it mean that is increases `um`'s context to include previous _questions_? Or does it mean that it's looking at the last `number` commands in my local shell's history file, and using that for context?

The request here would be to include previous _question_ context, so that I can say something like "the last command didn't give me what I wanted, can you try filtering out opt in regions?"

**Comment by jj-ctrlstack** [comment]
Yeah what you are requesting here seems very doable but it's different from the history context. Let me think about how we can have the previous questions injected as context as well (maybe as simple as pick from history?).
But, we should definitly clarify in the app or readme what the history context means. Just real quick here, the history context is meant to take the last few commands from the history as extra context. So for example you can do something like
```
$ curl http://localhost:8080/query\?query\=test
...
$ um use post and convert the params to json body

➜︎ ✨ curl -X POST -H 'Content-Type: application/json' -d '{"query": "test"}' http://localhost:8080/query
   💭️ don't see what you're looking for? try providing more context
```
(actual output btw)
However, I see it's not fully wired. So this one definitely need fixing as well.
